src/hooks/AllPaymentHooks.res

```diff
@@ -864,33 +864,12 @@ let useSavePaymentMethod = () => {
   let baseUrl = GlobalHooks.useGetBaseUrl()()
   let apiLogWrapper = LoggerHook.useApiLogWrapper()
   let (nativeProp, _) = React.useContext(NativePropContext.nativePropContext)
-  let (cardData, _) = React.useContext(CardDataContext.cardDataContext)
 
-  let (month, year) = Validation.getExpiryDates(cardData.expireDate)
-  let payment_method_data =
-    [
-      (
-        "card",
-        [
-          ("card_number", cardData.cardNumber->Validation.clearSpaces->JSON.Encode.string),
-          ("card_exp_month", month->JSON.Encode.string),
-          ("card_exp_year", year->JSON.Encode.string),
-        ]
-        ->Dict.fromArray
-        ->JSON.Encode.object,
-      ),
-    ]
-    ->Dict.fromArray
-    ->JSON.Encode.object
-
-  let body: PaymentMethodListType.redirectType = {
-    payment_method: "card",
-    client_secret: nativeProp.pmClientSecret->Option.getOr(""),
-    payment_method_data,
-  }
-
-  () => {
-    let paymentMethodId = nativeProp.paymentMethodManagementId->Option.getOr("")
+  (
+    ~body: PaymentMethodListType.redirectType
+  ) => {
+    let paymentMethodId =
+      String.split(nativeProp.clientSecret, "_secret_")->Array.get(0)->Option.getOr("")
```

Comment:
In payments flow, we extract the payment-id from client-secret in the hook itself. 
As implemented here: [useRedirectHook in AllPaymentHooks.res](https://gitkraken.dev/link/dnNjb2RlOi8vZWFtb2Rpby5naXRsZW5zL2xpbmsvci82MmY1ZTE4MDNkM2E0NDFhYmZlOTMyM2QyOGRmNDk1MmYxMWI3N2M2L2Yvc3JjL2hvb2tzL0FsbFBheW1lbnRIb29rcy5yZXM%2FdXJsPWdpdCU0MGdpdGh1Yi5jb20lM0FqdXNwYXklMkZoeXBlcnN3aXRjaC1jbGllbnQtY29yZS5naXQmbGluZXM9MzkyLTM5Mw%3D%3D?origin=gitlens). 

Also do we get any added advantage for extracting it at the top level?

Reply:
yes we have to extract at the top level because the logic may change "_secret_" also if someone uses the oss version if he makes any changes he don't have to look at multiple places, please create one more field in SDK types and while storing the client secret store the id as well

Reply:
done 👍